crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -158,13 +158,15 @@ impl TextureAtlas {
     }
 
     /// Add a sprite to the list of textures in the `TextureAtlas`
+    /// returns an index to the texture which can be used with `TextureAtlasSprite`
     ///
     /// # Arguments
     ///
     /// * `rect` - The section of the atlas that contains the texture to be added,
     /// from the top-left corner of the texture to the bottom-right corner
-    pub fn add_texture(&mut self, rect: Rect) {
+    pub fn add_texture(&mut self, rect: Rect) -> u32 {
```

Comment:
Taking a look at `get_texture_index` we are retuning the index as `usize`. I think we should do the same here.

Reply:
My preference would actually be to change get_texture_index to return a u32 too.  `TextureAtlasSprite` and the shader for it takes a u32.

Reply:
I think we should come up with a guideline to keep Bevy consistent. My opinion would be to use `usize` for indices and cast them to `u32` when passing them to shaders. [Initial responses on Discord](https://discord.com/channels/691052431525675048/692572690833473578/891380092868104213) suggest the same. Making all this consistent should be another PR thought, so I wouldn't block this one based on the inconsistency.

Reply:
For now, this needs to be u32 because we directly use the TextureAtlasSprite type when generating shader uniform buffers. In the renderer rework, this is no longer the case. And I agree that we should favor usize in user facing types for more natural indexing.

I think we should merge this as-is. Can you create another PR for the `pipelined-rendering` branch and make the relevant types in `bevy_sprites2` use usize?